node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1133,19 +1145,34 @@ async fn fetch_to_genesis(trusted_block: &Block, ctx: &ChainSyncContext<'_>) ->
 const MAX_HEADERS_BATCH_SIZE: u64 = 1024;
 
 // Fetches headers starting from `trusted_block` till the Genesis.
+// Returns the block header of the heighest switch block, if available.
 async fn fetch_headers_till_genesis(
     trusted_block: &Block,
     ctx: &ChainSyncContext<'_>,
-) -> Result<(), Error> {
+) -> Result<Option<BlockHeader>, Error> {
```

Comment:
Instead of returning it here I would keep this method's return type unchanged and fetch the Genesis era validators manually in the `fetch_to_genesis` body.